src/tribler/gui/core_manager.py

```diff
@@ -118,6 +125,7 @@ def start_tribler_core(self):
         connect(self.core_process.finished, self.on_core_finished)
         self._logger.info(f'Start Tribler core process {sys.executable} with arguments: {core_args}')
         self.core_process.start(sys.executable, core_args)
+        self.port_checker.setup_with_pid(self.core_process.processId())
```

Comment:
As it turns out, the reason it does not work correctly on my PC is venv. Venv creates a separate `python.exe` wrapper that calls the actual Python inside. So, the PID of the Core process that `QProcess` returns is the PID of the venv wrapper and not of the actual Core process. As most Tribler users use the compiled version of Tribler, they should not experience this problem. Nevertheless, we should be aware of this venv behavior.

On my machine, the PR is working, but not in the way it may be expected. As the actual PID of the Core process is different from what `QProcess` returns, the PortChecker cannot find the correct port, so it never calls the `EventRequestManager.update_port` method. Instead, the timeout of `EventRequestManager.connect_timer` forces the request with the default port value, which turns out to be the correct one.